### PR TITLE
Feature/use subchart to support multiple starrocks cluster

### PR DESCRIPTION
**Pull request by yandongxiao**

### CODE REVIEW

File: helm-charts/charts/kube-starrocks/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 apiVersion: v2
 name: kube-starrocks
-description: kube-starrocks collects Kubernetes manifests, starrocks combined with documentation and scripts to provide easy to operate end-to-end Kubernetes cluster deploy starrocks using the starrocks operator.
+description: kube-starrocks include tow subcharts, starrocks-operator and starrocks
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/Chart.yaml:3]:
```suggestion
description: kube-starrocks include two subcharts, starrocks-operator and starrocks
```

File: helm-charts/charts/kube-starrocks/Chart.yaml [hunk 2]
```diff
@@ -42,4 +44,4 @@ keywords:
 sources:
   - https://github.com/StarRocks/starrocks
 
-home: https://github.com/StarRocks/starrocks-kubernetes-operator
+home: https://github.com/StarRocks/starrocks-kubernetes-operator
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/Chart.yaml:47]: make sure your IDE append a new line to the end of the file. avoid change this EOL back and forth.

File: helm-charts/charts/kube-starrocks/charts/operator/templates/NOTES.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Thank you for installing {{ .Chart.Name }}-{{ .Chart.Version }}
+
+To create a StarRocks cluster refer to the following documentation.
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/charts/operator/templates/NOTES.txt:3]:
```suggestion
Please refer to the following document to create a StarRocks cluster.
```

File: helm-charts/charts/kube-starrocks/charts/operator/values.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+global:
+  rbac:
+    # if set true, the clusterrole, clusterrolebinding, role, rolebinding, serviceaccount resources will be created for
+    # operator.
+    create: true
+  # set the starrockscluster CRD version. From operator v1.6 we have supported v1 CRD and as default CRD version, but
+  # the v1alpha1 version is also supported.
+  # supported value: v1, v1alpha1.
+  crdVersion: v1
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply helm-charts/charts/kube-starrocks/charts/operator/values.yaml:13]: this filed is only referenced by helm chart in one place. And I want to delete it from values.yaml.

File: helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+    tag: 3.0-latest
+  # add annotations for fe pods. For example, if you want to config monitor for datadog, you can config the annotations.
+  annotations: {}
+  # If runAsNonRoot is true, the container is run as non-root user.
+  # The userId will be set to 1000, and the groupID will be set to 1000.
+  runAsNonRoot: false
+  # specify the service name and port config and serviceType
+  # the service type refer https://kubernetes.io/docs/concepts/services-networking/service/#publishing-services-service-types
+  service:
+    # the fe service type, only supported ClusterIP, NodePort, LoadBalancer, ExternalName
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:58]: 1. I think we can support `ClusterIP, NodePort, LoadBalancer` and remove the `externalName`.
2. there are `ports` filed in CRD, but does not implement it in controller.  chart did not use it.

File: helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml [hunk 6]
```diff
@@ -0,0 +1,406 @@
...
+starrocksCluster:
+  # the name of starrockscluster cluster, if not set, the chart name will be used.
+  name: ""
+  # the namespace of starrockscluster cluster, if not set, the release namespace will be used.
+  namespace: ""
+  # annotations for starrocks cluster.
+  # there are few special annotations:
+  #   1. when key is "app.starrocks.fe.io/restart", and value is "restart", the fe component will be restarted.
+  #   2. when key is "app.starrocks.be.io/restart", and value is "restart", the be component will be restarted.
+  #   3. when key is "app.starrocks.cn.io/restart", and value is "restart", the cn component will be restarted.
```
[the remark this answers is not in the document]
Comment (yandongxiao) [reply helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:41]: The Annotation will be changed when controller do `sync` status.
```go
	stValue := st.Annotations[string(srapi.AnnotationBERestartKey)]
	srcValue := src.Annotations[string(srapi.AnnotationBERestartKey)]
	if (srcValue == string(srapi.AnnotationRestart) && stValue == string(srapi.AnnotationRestarting)) ||
		(srcValue == string(srapi.AnnotationRestarting) && stValue == string(srapi.AnnotationRestartFinished)) {
		src.Annotations[string(srapi.AnnotationBERestartKey)] = stValue
	}
```
Next time, when user upgrade helm chart, where will trigger fe/be/cn to restart. That is a Bug!

And I do not think we really need `xxx/restart` annotations. If user want to restart fe/be/cn, they just update or add a annotation by themself.

File: helm-charts/charts/kube-starrocks/Chart.yaml [hunk 7]
```diff
@@ -1,6 +1,7 @@
 apiVersion: v2
 name: kube-starrocks
-description: kube-starrocks collects Kubernetes manifests, starrocks combined with documentation and scripts to provide easy to operate end-to-end Kubernetes cluster deploy starrocks using the starrocks operator.
+description: kube-starrocks include two subcharts, starrocks-operator and starrocks
```
Comment (kevincai) [inline helm-charts/charts/kube-starrocks/Chart.yaml:3]:
```suggestion
description: kube-starrocks includes two subcharts, starrocks-operator and starrocks
```

File: helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (kevincai) [reply helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:41]: let's remove these annotation support in a separate PR

File: helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (kevincai) [reply helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:62]: I think if user provides `ports` configuration, it still works. We need to make sure the behavior is consistent and predictable.

[the remark this answers is not in the document]
Comment (kevincai) [reply helm-charts/charts/kube-starrocks/charts/starrocks/values.yaml:62]: b.t.w. ExternalName can be removed.

**Review by kevincai — APPROVED** [review]

**Pull request merged**